src/expr/node_manager.h

```diff
@@ -442,7 +434,7 @@ class NodeManager
    * would lead to memory leaks. Thus TypeNode are given a DatatypeIndexConstant
    * which is used as an index to retrieve the DType via this call.
    */
-  const DType& getDTypeForIndex(size_t index) const;
+  const DType& getDType(size_t index) const;
```

Comment:
I'd prefer keeping the original name here, since it is indicative of the use of an index for the lookup.  This makes the method more descriptive and distinguishes it from typical uses e.g. TypeNode::getDType.